### DISCUSSION TITLE
Jobs started in 1 minute before planned time

### DISCUSSION THREAD

**Discussion by igorbljahhin** [jobrunr/jobrunr#770] [General]
Hello,
I am observing that JobRunr starts recurring jobs in 1 minute before planned time. For example, my job was planned to be executed every day at 10:45 UTC.

The job writes messages into the log when it is scheduled and when it starts:

> 2023-05-22T10:54:05.999Z
> Scheduling recurring job "ADMARITIME_transactions" to run every day at 10:55 using cron expression "0 55 10 * * *"...
> 
> 2023-05-22T10:54:06.022Z
> Recurring job "ADMARITIME_transactions" for the report TRANSACTIONS has been scheduled to run every day at 10:55 (UTC) using cron expression "0 55 10 * * *"!
> 
> 2023-05-22T10:54:19.467Z
> The report TRANSACTIONS will be generated and saved into the file /mnt/azure/bora-systems/landing/transaction/Transaction_20230521.csv

I would expect that job is started at 10:55 or a couple seconds later.

The job is scheduled using this call:

```
        final String cronExpression = "0 " + request.getScheduleAt().getMinute() + " " + request.getScheduleAt().getHour() + " * * *";

        log.info("Scheduling recurring job \"{}\" to run every day at {} using cron expression \"{}\"...", recurringJobDescriptionId, request.getScheduleAt(), cronExpression);

        // schedule job
        jobScheduler.scheduleRecurrently(recurringJobDescriptionId, cronExpression, () -> reportService.generate(JobContext.Null, "ADMARITIME", "TRANSACTIONS", recurringJobDescriptionId));
```

Jobrunr version is 5.3.3. The application is running in Docker container. I checked its internal time and it is correct and the container's time is synchronized with the time of NTP server.

What might be a reason for this strange case?

**Comment by rdehuyss** [comment]
Feel free to join this '[Meet the geek' session](https://www.jobrunr.io/en/blog/2023-05-24-meet-the-geek/) where I'll try to answer your question.

**Comment by igorbljahhin** [comment]
You were right, the job something around 15 seconds ahead of the planned time. I made workaround suggested by you, now our jobs are planned with 30 seconds delay, so they start more or less on time.
